src/main/java/com/qaprosoft/carina/demo/mobile/gui/mfp/pages/android/CommonPage.java

```diff
@@ -22,4 +35,19 @@ public boolean isUserLoggedIn(String username) {
         return true;
     }
 
+    @Override
+    public AbstractPage clickBottomNavigatorButton(BottomNavigatorButtons button) {
+        switch (button)
+        {
+            case DASHBOARD: dashboardButton.click();
+                return initPage(getDriver(), DashboardPageBase.class);
+            case DIARY: diaryButton.click();
+                return initPage(getDriver(), DiaryPageBase.class);
+            case NEWSFEED: newsfeedButton.click();
+                return initPage(getDriver(), NewsfeedPageBase.class);
+            case PLANS: plansButton.click();
+                return initPage(getDriver(), PlansPageBase.class);
+            default: return null;
```

Comment:
bad practice! Perfect solution - universal xpath like //*[@resource-id = 'com.myfitnesspal.android:id/text' and contains(@text,'%s')]" where we insert our enum option, using method from enum, e.g.: EnumName.getMenuOptionName. Investigate, pls, if it's possible. if it does - in case we got just page, in default section - click on desired element. if it's not possible - leave it this way, but in default section put logger, like: LOGGER.info(String.format("[ More Menu ] Navigated to '%s' page.", button.getButton()));